File: PROGRAMMERS/큐/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    # deque로 튜플 형식으로 집어넣음
    queue = deque([(index, priority) for index, priority in enumerate(priorities)])
    
    while len(queue) > 0:
        # 프로세스 하나 뺌
        temp = queue.popleft()
        # 우선순위 높은거 있으면 다시 넣고
        if any(temp[1]<num[1] for num in queue):
            queue.append(temp)
        else:
            # 높은 거 없음
            answer += 1
            # 위치 비교
            if temp[0] == location:
                return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        // 우선순위가 높은 프로세스
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int priority : priorities){
            pq.add(priority);
        }
        
        while(!pq.isEmpty()){
            for(int i=0; i<priorities.length; i++){
                // 기존 큐 순서대로 하기
                
                // 가장 큰 우선순위랑 지금 값이랑 같은지
                if(pq.peek() == priorities[i]){
                    answer++;
                    
                    // 지금 찾는 위치가 아니면
                    if(i == location){
                        return answer;
                    }
                
                    pq.poll();
                }
             }
        }
  
        return answer;
    }
}